src/rdkafka_request.c

```diff
@@ -1878,6 +1878,49 @@ void rd_kafka_ListGroupsRequest_versioned(rd_kafka_broker_t *rkb,
         rd_kafka_broker_buf_enq_replyq(rkb, rkbuf, replyq, resp_cb, opaque);
 }
 
+/**
+ * Send DescribeGroupsRequest supporting a versioned request.
+ */
+void rd_kafka_DescribeGroupsRequest_versioned(rd_kafka_broker_t *rkb,
+                                              const char **groups,
+                                              int group_cnt,
+                                              rd_kafka_replyq_t replyq,
+                                              rd_kafka_resp_cb_t *resp_cb,
+                                              void *opaque) {
+        rd_kafka_buf_t *rkbuf;
+        int i;
+        size_t size;
+        int16_t api_version;
+        size_t groups_arraycnt;
+
+        api_version = rd_kafka_broker_ApiVersion_supported(
+            rkb, RD_KAFKAP_DescribeGroups, 0, 4, NULL);
+
+        size = 4 /* rd_kafka_buf_write_arraycnt_pos */ +
+               1 /* IncludeAuthorizedOperations */ + 1 /* tags */;
+        for (i = 0; i < group_cnt; i++) {
+                size += 2 + strlen(groups[i]);
+        }
+
+        rkbuf = rd_kafka_buf_new_flexver_request(rkb, RD_KAFKAP_DescribeGroups,
+                                                 1, size, rd_false);
+
+        /* write Groups */
+        groups_arraycnt = rd_kafka_buf_write_arraycnt_pos(rkbuf);
+        rd_kafka_buf_finalize_arraycnt(rkbuf, groups_arraycnt, group_cnt);
```

Comment:
It's a bit mixed in the code, but most new protocol code uses the AK CamelCase for protocol fields, so,  GroupsArrayCnt, etc.